### ISSUE TITLE
Do not return 40x for `releaseHold` on `Running` workflow

### ISSUE THREAD

**Issue by mcovarr** [broadinstitute/cromwell#4983]
Our Green friends are regularly seeing cases where Cromwell reports a workflow's `status` as `On Hold` but follow-on calls to `releaseHold` return a 403 status. This appears to be happen because `status` queries go to the potentially laggy workflow summary table but `releaseHold` goes straight to the workflow store. Cromwell [returns 403](https://github.com/broadinstitute/cromwell/blob/develop/engine/src/main/scala/cromwell/webservice/routes/CromwellApiService.scala#L148) if the workflow for which `releaseHold` is invoked is [in the workflow store but not in `On Hold` or `Submitted`](https://github.com/broadinstitute/cromwell/blob/develop/engine/src/main/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStore.scala#L140) status. 

403s seem like a strange status to return in this case. If the thought was "the workflow has to be on hold to be able to release the hold" then we would also return 403 for workflows found in `Submitted`, but we don't; those get a 200. Since the status queries will always be laggy to some degree it seems to make more sense to interpret the endpoint as "put this workflow in a runnable state", where finding the workflow already in a `Running` state would not be cause for alarm.

**Comment by cjllanwarne** [comment]
Is this the sort of scheme you mean? ie considering the operation as idempotent if you ask to release twice? 
* `releaseHold` on a held workflow => 201
* `releaseHold` on a running workflow => 200

**Comment by mcovarr** [comment]
@cjllanwarne yes. This won't protect us from returning 40x for workflows that are not in the store that may have legitimately gone terminal, but at least this won't produce errors for workflows that are obviously OK.
